mlir/test/Dialect/Tosa/canonicalize.mlir

```diff
@@ -47,6 +47,22 @@ func.func @cast_fold_double(%arg0: tensor<?x1xf32>) -> tensor<?x1xi8> {
   return %1 : tensor<?x1xi8>
 }
 
+// CHECK-LABEL: @cast_fold_same_type
+func.func @cast_fold_same_type(%arg0: tensor<?x1xf32>) -> tensor<?x1xf32> {
+  // CHECK: return %arg0 : tensor<?x1xf32>
```

Comment:
What happens when I give `2^17` into this function? Will it still return the same result after the fold?

Reply:
yes, just talked to @TinaAMD , I am closing this.